### ISSUE TITLE
Characters in URLs should not be interpreted by the markdown parser

### ISSUE THREAD

**Issue by karlcow** [webcompat/webcompat.com#1171]
When someone creates an issue with strange characters in the URL, these should not be interpreted as formatting characters.

![capture d ecran 2016-09-06 a 10 24 12](https://cloud.githubusercontent.com/assets/505230/18259192/473a6c94-741c-11e6-8ffd-3f4b20fc18b6.png)

**Comment by karlcow** [comment]
The URI used for this report is in
```
http://example.com/foobar&amp=**blah_boo**&what=`coucou`
```

**Comment by devjayati** [comment]
i don't know if this would be of help, but comparing different markdown implementations [here](http://johnmacfarlane.net/babelmark2/?text=http%3A%2F%2Fexample.com%2Ffoobar%26amp%3D**blah_boo**%26what%3D%60coucou%60) shows Marked 0.2.6 does not read it as formatting characters

**Comment by zoepage** [comment]
@karlcow 

The URL is passed in as <a href="http://example.com/foobar&amp;amp=" rel="nofollow">http://example.com/foobar&amp;amp=</a> already, so it's not just about display of the URL, but a step before that, right and also just in our interface.
<img width="730" alt="screen shot 2017-04-03 at 10 39 15 am" src="https://cloud.githubusercontent.com/assets/562354/24601803/990bcbb6-185a-11e7-86e1-e0efebdef5bf.png">

Markdown in general looks fine plus the URL is correct.
<img width="706" alt="screen shot 2017-04-03 at 10 39 09 am" src="https://cloud.githubusercontent.com/assets/562354/24601802/990872fe-185a-11e7-888c-24fca0e32642.png">



As you created the issue, do you have a suggestion for a fix?

**Comment by karlcow** [comment]
Yes the markdown parser should not interpret the URL. It's what the issue is about. :) 
To fix it, it is possible to either escape the markdown markup, but probably better take the opportunity of the `target` issue

**Comment by miketaylr** [comment]
> Probably processing the payload with a series of different filters in their own individual function so we can remove/activate them at will (python side).

That sounds nicer than hacking the markdown parser.

**Comment by karlcow** [comment]
Same type of issues for https://webcompat.com/issues/14182

<img width="687" alt="capture d ecran 2017-12-12 a 14 31 35" src="https://user-images.githubusercontent.com/505230/33907032-662d9786-df49-11e7-9826-47372d51d43b.png">
